service/client.go

```diff
@@ -114,6 +114,8 @@ type Client interface {
 
 	// ListGoroutines lists all goroutines.
 	ListGoroutines(start, count int) ([]*api.Goroutine, int, error)
+	// ListGoroutinesWithFilter lists goroutines matching the filters
+	ListGoroutinesWithFilter(start, count int, filters []api.ListGoroutinesFilter, groupBy api.GoroutineField, groupByArg string, maxGroupExamples, maxGroups int) ([]*api.Goroutine, []api.GoroutineGroup, int, bool, error)
```

Comment:
I'd prefer a option struct argument instead of this large list of arguments.

Reply:
I think that it would be complicated to capture every possibility on a struct, for example:

```
goroutines -w label x -wo label x=b
```

will show all goroutines with a label with key 'x' that do not have value 'b'. Combining multiple with/without conditions on the same location could also be useful.

Reply:
I'm not sure I understand, would this be:

```
type ListGoroutinesOptions struct {
    Filters []ListGoroutinesFilter
    ListGoroutinesGroupingOptions
}

type ListGoroutinesGroupingOptions struct {
    GroupBy GoroutineField
    GroupByArg string
    MaxGroupExamples int
    MaxGroups int
}
```

?

Reply:
I was thinking out loud and suggesting either what you mentioned _or_ just doing `ListGoroutinesGroupingOptions`, but since both filter and group are optional I think it makes sense to put all arguments related to those features in an optional struct. That way if the caller does not want to use those features they only have to pass through a single `nil` for that struct instead of the zero value of 4-5 arguments.

Reply:
Ok, done.